protocol-units/da-sequencer/node/src/server.rs

```diff
@@ -177,27 +181,44 @@ impl DaSequencerNodeService for DaSequencerNode {
 		Ok(tonic::Response::new(Box::pin(output) as Self::StreamReadFromHeightStream))
 	}
 
-	/// Batch write blobs.
+	/// Batch write blobs
 	async fn batch_write(
 		&self,
 		request: tonic::Request<BatchWriteRequest>,
 	) -> std::result::Result<tonic::Response<BatchWriteResponse>, tonic::Status> {
 		let batch_data = request.into_inner().data;
-		let batch = match crate::batch::deserialize_full_node_batch(batch_data).and_then(
-			|(public_key, signature, bytes)| {
-				validate_batch(DaBatch::<RawData>::now(public_key, signature, bytes))
-			},
-		) {
-			Ok(batch) => batch,
-			Err(err) => {
-				tracing::warn!(error = %err, "Invalid batch send, verification / validation failed.");
-				return Ok(tonic::Response::new(BatchWriteResponse { answer: false }));
+
+		// Try to deserialize the batch
+		let (public_key, signature, bytes) =
```

Comment:
Instead of matching on the error here, use `map_or_else` for a more concise expression.

Reply:
hmm I mean't something like this, I think it's possible?
```rust
let (public_key, signature, bytes) = crate::batch::deserialize_full_node_batch(batch_data)
    .map_or_else(
        |err| {
            tracing::warn!("Invalid batch send: deserialization failed: {err}");
            return Ok(tonic::Response::new(BatchWriteResponse { answer: false }));
        },
        |res| res,
    );
    ```
    no `match` required.